test/test_out_s3.rb

```diff
@@ -109,6 +109,14 @@ def test_configure_with_mime_type_lzo
       assert(e.is_a?(Fluent::ConfigError))
     end
 
+    def test_configure_with_mime_type_zstd
+      conf = CONFIG.clone
+      conf << "\nstore_as zstd\n"
+      d = create_driver(conf)
+      assert_equal 'zst', d.instance.instance_variable_get(:@compressor).ext
+      assert_equal 'application/x-zst', d.instance.instance_variable_get(:@compressor).content_type
+    end
```

Comment:
We can add a test for the `level` option like this.

```suggestion
    data('level default' => nil,
         'level 1' => 1)
    def test_configure_with_mime_type_zstd(level)
      conf = CONFIG.clone
      conf << "\nstore_as zstd\n"
      conf << "\n<compress>\nlevel #{level}\n</compress>\n" if level
      d = create_driver(conf)
      assert_equal 'zst', d.instance.instance_variable_get(:@compressor).ext
      assert_equal 'application/x-zst', d.instance.instance_variable_get(:@compressor).content_type
      assert_equal (level || 3), d.instance.instance_variable_get(:@compressor).instance_variable_get(:@compress_config).level
    end
```